fs/fuse/file.c

```diff
@@ -149,23 +217,42 @@ struct fuse_file *fuse_file_open(struct fuse_mount *fm, u64 nodeid,
 	ff->open_flags = FOPEN_KEEP_CACHE | (isdir ? FOPEN_CACHE_DIR : 0);
 	if (open) {
 		struct fuse_open_out outarg;
-		int err;
+		int err = -ENOSYS;
+
+		if (inode && fc->compound_open_getattr) {
```

Comment:
For upstream we still have an issue here - the application might actually not want to do open+getattr. I.e. let's say libfuse supports compounds and can handle open+attr, the application might say "I don't need it, it is just overhead". I wonder if we need a new FUSE_INIT flag for that part, as ENOSYS is already taken.